TotalCrossSDK/src/main/java/totalcross/json/JSONFactory.java

```diff
@@ -10,48 +10,55 @@
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
 /**
-    The JSONFactory class helps converting json objects into Java objects, using reflection.
-
-    Some examples:
-
-    <pre>
-    class Car 
-    {
-       private int id;
-       private String description;
-
-       public int getId()
-       {
-          return id;
-       }
-       public void setId(int id)
-       {
-          this.id = id;
-       }
-       public String getDescription()
-       {
-          return description;
-       }
-       public void setDescription(String description)
-       {
-          this.description = description;
-       }
-    }
-    </pre>
-
-    You can retrieve a new Car object using:
-    <pre>
-    Car cc = JSONFactory.parse("{\"carro\":{\"id\":-1,\"descricao\":\"GOL\"}}", Carro.class);
-    </pre>
-
-    You may also retrieve a list or an array. See the JSONSample in the TotalCrossAPI.
+ * The JSONFactory class helps converting json objects into Java objects, using
+ * reflection.
+ * 
+ * Some examples:
+ * 
+ * <pre>
+ * class Car {
+ *   private int id;
+ *   private String description;
+ * 
+ *   public int getId() {
+ *     return id;
+ *   }
+ * 
+ *   public void setId(int id) {
+ *     this.id = id;
+ *   }
+ * 
+ *   public String getDescription() {
+ *     return description;
+ *   }
+ * 
+ *   public void setDescription(String description) {
+ *     this.description = description;
+ *   }
+ * }
+ * </pre>
+ * 
+ * You can retrieve a new Car object using:
+ * 
+ * <pre>
+ * Car cc = JSONFactory.parse("{\"carro\":{\"id\":-1,\"descricao\":\"GOL\"}}", Carro.class);
+ * </pre>
+ * 
+ * You may also retrieve a list or an array. See the JSONSample in the
+ * TotalCrossAPI.
  */
 public class JSONFactory {
-  public static <T> List<T> asList(String json, Class<T> classOfT) throws InstantiationException,
-      IllegalAccessException, IllegalArgumentException, InvocationTargetException, JSONException, ArrayIndexOutOfBoundsException, NoSuchMethodException, SecurityException {
+  private static Map<Class<?>, Map<String, Method>> classes = new HashMap<>();
```

Comment:
Using this as a source of cache, one should be aware that it isn't thread safe **OR** the thread safety must be provided by the implementor